src/averageAges.js

```diff
@@ -56,6 +73,30 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  let filteredChildren = people.filter(person => people.find(
+    mother => mother.name === person.mother
+  ));
+
+  if (onlyWithSon) {
+    filteredChildren = filterBySex(filteredChildren, 'm');
+  }
+
+  const ageDifference = filteredChildren.reduce((prev, curr) =>
+    (prev + curr.born - people.find(person =>
+      person.name === curr.mother).born), 0);
+
+  return ageDifference / filteredChildren.length;
+}
+
+function filterBySex(people, sex) {
+  return people.filter(person => person.sex === sex);
+}
+
+function averageAge(filteredArray) {
+  const sumOfAge = filteredArray.map(person => person.died - person.born)
+    .reduce((prev, curr) => prev + curr);
```

Comment:
You can use only reduce here, just take object instead of number here, and add to accumulator difference between born and died properties